tests/common/hyperparameter_tuner/test_hyperparameter_tuner.py

```diff
@@ -0,0 +1,663 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any, Dict, List, Tuple
+
+import pytest
+
+from nncf.quantization.algorithms.hyperparameter_tuner.algorithm import apply_combination
+from nncf.quantization.algorithms.hyperparameter_tuner.algorithm import create_combinations
+from nncf.quantization.algorithms.hyperparameter_tuner.algorithm import filter_combinations
+from nncf.quantization.algorithms.hyperparameter_tuner.algorithm import find_best_combination
+from nncf.quantization.algorithms.hyperparameter_tuner.algorithm import trim_zeros
+
+CombinationKey = Tuple[int, ...]
+TrimedCombinationKey = Tuple[int, ...]
+Combination = Dict[str, Any]
+
+
+# =========================================================
+# TEST: Create combinations
+# =========================================================
+
+
+@pytest.mark.parametrize(
+    "param_settings,expected_combinations",
+    [
+        # one_parameter
+        (
+            # param_settings
+            {"x": [0, 1, 2]},
+            # expected_combinations
+            {
+                # x
+                (0,): {},
+                (1,): {"x": 0},
+                (2,): {"x": 1},
+                (3,): {"x": 2},
+            },
+        ),
+        # two_parameters
+        (
+            # param_settings
+            {"x": [0, 1, 2], "y": [True, False]},
+            # expected_combinations
+            {
+                # x
+                (0,): {},
+                (1,): {"x": 0},
+                (2,): {"x": 1},
+                (3,): {"x": 2},
```

Comment:
These combinations contain in the `# x, y` below. Why do we need it?
If we need it, why don't we have `# y` combinations then?

Reply:
Implementation was changed. Please take a look again.